subprojects/functional/src/main/java/org/gradle/internal/Either.java

```diff
@@ -48,14 +48,20 @@
     public abstract Optional<R> getRight();
 
     /**
-     * Map either case.
+     * Map the left side.
+     *
+     * @see #mapRight
+     * @see #fold
      */
-    public abstract <U, V> Either<U, V> map(Function<? super L, ? extends U> l, Function<? super R, ? extends V> r);
+    public abstract <U, V> Either<U, V> mapLeft(Function<? super L, ? extends U> f);
```

Comment:
I guess you don't need these either, and instead can use `Either.getLeft().map()` etc. I think it would make sense to keep it simple and remove these.

Reply:
No strong feelings. I think that while not essential (only `fold` is in the end) they capture a common enough pattern, at the same time, they are not being used.